### ISSUE TITLE
Warnings vs Errors

### ISSUE THREAD

**Issue by shellscape** [lesshint/lesshint#44]
Given the discussion around https://github.com/lesshint/lesshint/issues/43 errors being swallowed, I'd like to raise the question of warnings vs errors. jshint plops every result into an 'errors' object, but has a code on every result that differentiates error from warning. In jshint's case, it has `W` for warning, `I` for info, and `E` for error. I'm not suggesting lesshint follow suit, as I've never been incredibly keen on using single letters for error codes, but it would be a great addition to the project to differentiate.

Consider this lesshint gulp reporter: https://github.com/sagikazarmark/gulp-lesshint-stylish, which is transforming the results and marking everything back from lesshint as a warning.

Ideally everything that lesshint is returning as a result at this moment, sans parse error support, should be a warning. Once parse error support is added, those results should be errors. I'm not sure if `info` has any use here.

**Comment by jwilsson** [comment]
I totally agree and have actually been thinking about this myself.

Another feature that could follow from this is the ability to specify the severity of a rule violation. Everything defaults to `warning` but if desired, mark some rule violations as `errors` and exit with a different code etc.

@kokarn thoughts, ideas?

**Comment by shellscape** [comment]
:+1:  to the ability to override severity

**Comment by kokarn** [comment]
I think @jwilsson is spot on with this and that we all agree that's something that could be useful.

I don't think we should follow jshint tho, add warnings/info to an error object seems a bit backwards and the one letter codes just seems a bit stupid. 
@shellscape Do you know why they do single letters?

I think that we should either return a single object with all warnings/error/info on it or a single object with three objects, each containing it's own type. 

The more i think about it i like the second alternative a bit more, would be easier to enumerate, could enable handling of specific types differently without having to loop over everything and so on.

**Comment by shellscape** [comment]
@kokarn no clue why they used single letters - if I had to guess it was left over from jslint or just a style choice. I'm personally OK with an array of result objects, each with their 'state' property. Splitting that up into `{ warnings: [], errors: [] }` is a matter of stylistic preference at that point, or another way to map the data. 

I've started work on this on our fork and will submit PRs along the way.
